### PR TITLE
Duplicate scene

### PR DESCRIPTION
**Pull request by callemand**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [x] If your changes affects code, did your write the tests?
- [x] Are tests passing? (`npm test` on both front/server)
- [x] Is the linter passing? (`npm run eslint` on both front/server)
- [x] Did you run prettier? (`npm run prettier` on both front/server)
- [x] If you are adding a new features/services, did you run integration comparator? (`npm run compare-translations` on front)
- [x] Did you test this pull request in real life? With real devices? If this development is a big feature or a new service, we recommend that you provide a Docker image to [the community](https://community.gladysassistant.com/) for testing before merging.
- [x] If your changes modify the API (REST or Node.js), did you modify the API documentation? (Documentation is based on comments in code)
- [ ] If you are adding a new features/services which needs explanation, did you modify the user documentation? See [the GitHub repo](https://github.com/GladysAssistant/v4-website) and the [website](https://gladysassistant.com).
- [ ] Did you add fake requests data for the demo mode (`front/src/config/demo.js`) so that the demo website is working without a backend? (if needed) See [https://demo.gladysassistant.com](https://demo.gladysassistant.com).

NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open.

### Description of change

The purpose of this PR is a proposal to add scene duplication on edit scene page.

Screenshot 2022-12-12 at 19 52 49

Screenshot 2022-04-15 at 14 40 15

### CODE REVIEW

File: front/src/config/i18n/en.json
Comment (Pierre-Gilles) [inline front/src/config/i18n/en.json]: This translation is very long (we can't see the full text). "duplicate" should be "duplicated"

Why not simply: 

> Enter a new name

?

File: front/src/config/i18n/fr.json
Comment (Pierre-Gilles) [inline front/src/config/i18n/fr.json]: Title is too long for a placeholder, we can't read it: 

Screenshot 2022-12-16 at 11 33 57

Why not:

> "Entrez un nouveau nom"

If you want to add an explanation, it should be added outside of the placeholder

File: server/test/lib/scene/scene.duplicate.test.js
Comment (Pierre-Gilles) [inline server/test/lib/scene/scene.duplicate.test.js]: To avoid having to disable eslint for this, you can do `.not.to.equal(undefined)` for example

File: server/test/lib/scene/scene.duplicate.test.js
Comment (Pierre-Gilles) [inline server/test/lib/scene/scene.duplicate.test.js:17]: Can you add a bit more checks? What about the triggers? What about the actions?

It should be strictly verified with a .deep.equal() imo :)

File: server/lib/scene/scene.duplicate.js
Comment (Pierre-Gilles) [inline server/lib/scene/scene.duplicate.js]: Just a small feedback, maybe it could have been wise here simply to call: 

```
return this.create(newScene);
```

? 

This way we avoid duplicating code, and if we add new things to the scene.create function, we avoid having to copy-paste code here :)

**Review by Pierre-Gilles — CHANGES_REQUESTED** [review]
Thank you so much for this PR 🙏 

This is great work and it works very well on my machine 😄 

I just have a few small feedbacks, it's really nothing.

After that, I'm good to merge!

File: server/lib/scene/scene.duplicate.js
Comment (callemand) [reply server/lib/scene/scene.duplicate.js]: Yes, you are completely right 👍

File: front/src/routes/scene/duplicate-scene/index.js
Comment (Pierre-Gilles) [inline front/src/routes/scene/duplicate-scene/index.js]: Oups sorry I didn't see this during last review, can you use async/await ? 

We try to avoid using .then because it leads to long chain of .then

**Review by Pierre-Gilles — CHANGES_REQUESTED** [review]
Thanks for your changes it's perfect now 🙏 

I just found a last tiny 1 line change that I didn't see last time, sorry 😂

I merge this right after the change! Thanks!

File: front/src/routes/scene/duplicate-scene/index.js
Comment (callemand) [reply front/src/routes/scene/duplicate-scene/index.js]: I prefer the await too :)

**Review by Pierre-Gilles — APPROVED** [review]
Thanks for the changes, it's good for me ! ✅

**Pull request merged**
